Add recent release sort option to the website

AhmedBasem20 opened the pull request.

unkcpz commented on aiida-registry-app/src/Components/MainIndex.jsx:
```diff
@@ -55,13 +55,31 @@ function MainIndex() {
+          if (!pluginA.metadata.release_date && !pluginB.metadata.release_date) {
+           return 0; // Both plugins have no release date, keep them in the current order
+          } else if (!pluginA.metadata.release_date) {
+            return 1; // Only pluginB has a release date, so pluginA should come first
```
```suggestion
            return 1; // Only pluginB has a release date, put pluginB higher ranking than pluginA.
```

unkcpz commented on aiida-registry-app/src/Components/MainIndex.jsx:
```diff
@@ -55,13 +55,31 @@ function MainIndex() {
         sortedPlugins = plugins;
       }
+      else if (option == 'release') {
+        //Sort plugins by the recent release date
```
```suggestion
        // Sort plugins by the recent release date, the newer the release date the larger the value, and the higher ranking it get. Sorting in descending order by date. 
```

unkcpz commented on aiida-registry-app/src/Components/MainIndex.jsx:
```diff
@@ -55,13 +55,31 @@ function MainIndex() {
+          } else if (!pluginA.metadata.release_date) {
+            return 1; // Only pluginB has a release date, so pluginA should come first
+          } else if (!pluginB.metadata.release_date) {
+            return -1; // Only pluginA has a release date, so pluginB should come first
```
```suggestion
            return -1; // Only pluginA has a release date, put pluginA higher ranking than pluginB.
```

unkcpz reviewed (changes requested):
Thanks! I resolve the conflict after https://github.com/aiidateam/aiida-registry/pull/255. I think it is all good to go!
Just a nitpick on the comments of the code, I hope it will make the sorting order more clear, do you agree?

unkcpz commented:
I'll merge `gsoc` to `main` after this one.

AhmedBasem20 commented:
Thanks @unkcpz I corrected the comments you mentioned, and I also moved the code to a separate function `sortByRelease()`. And in `useEffect` I replaced the call to `sortByCommits()` with a call to `handleSort()`.
This way if we want to change the default sort option at any time we just change that line: https://github.com/aiidateam/aiida-registry/blob/82ab0f72205e89e0ca6cd3adfc370ae2eb405ca2/aiida-registry-app/src/Components/MainIndex.jsx#L19

unkcpz reviewed (approved):
Very clear, thanks!
